keel-core/src/main/kotlin/com/netflix/spinnaker/keel/verification/VerificationRunner.kt

```diff
@@ -44,6 +52,33 @@ class VerificationRunner(
     }
   }
 
+  private fun anyPendingVerificationsForPreviousVersions(context: VerificationContext) =
+    verificationRepository
+      .pendingInEnvironment(context.deliveryConfig, context.environmentName)
+      // only consider other versions, we'll handle verifications for the version in context later
+      .filterNot { it.context.version == context.version }
+      // get the latest status by re-evaluating each one (which will update in the database)
+      .associateWith { it.context.latestStatus(it.verification) }
+      // we're only concerned with anything that is still running after we have re-evaluated it
+      .filter { (_, status) -> status == PENDING }
```

Comment:
This is a slight change as we'd only have been looking at verifications that were `PENDING` and they can't go back from `PENDING` to `NOT_EVALUATED`.